cluster/cluster.go

```diff
@@ -563,3 +571,116 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
 	gplog.FatalOnError(err)
 	return segConfigs
 }
+
+/*GetSegmentConfigurationFromFile to parse gpsegconfig_dump or similar file to retrieve segment configuration information.
+gpsegconfig_dump file gets created at $COORDINATOR_DATA_DIR/gpsegconfig_dump by fts process
+The gpsegconfig_dump file follows a structured format, as illustrated in the example below:
+1 -1 p p n u 6000 localhost localhost /data/temp1
+2 0 p p n u 6002 localhost localhost /data/temp2
+3 1 p p n u 6003 localhost localhost /data/temp3
+4 2 p p n u 6004 localhost localhost /data/temp4
+
+Recommended use of the api is to get populate the gpdbArray when database is down.
+if the database is up use GetSegmentConfiguration()/MustGetSegmentConfiguration()  API
+
+input:
+confFile - absolute path of the gpsegconfig_dump or similar structured file
+(e.g. /Users/shrakesh/workspace/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/gpsegconfig_dump)
```

Comment:
with context to https://github.com/greenplum-db/gp-common-go-libs/pull/85#discussion_r1448329077 as well.
COORDINATOR_DATA_DIRECTORY we can get it from the env variable.  with this API it is the caller's responsibility to provide a valid gpsegconfig_dump or similar file path. 

in the previous commit, we are appending the path inside the API to get the gpsegconfig_dump file path. but after the suggestion on PR I also agree that the providing file path as a user argument will keep the API use consistent. 

Another option we have is that we can have a supporting API to get the gpsegconfig_dump file path. 

please let me know your opinion.

Reply:
IMO the calling function shouldn't have to know about `gpsegconfig_dump` since they only care about getting the configuration details when the database is offline. So I think the calling utility can just pass the coordinator/master data directory.

Now that I'm thinking about it, maybe it's not so bad if the function doesn't accept anything and the utility just figures out the env variable itself.  Need to think about this a bit more